File: gtest/expr/LiteralTest.cc

/// @file LiteralTest.cc
/// @brief LiteralTest の実装ファイル
/// @author Yusuke Matsunaga (松永 裕介)
///
/// Copyright (C) 2017 Yusuke Matsunaga
/// All rights reserved.


#include "gtest/gtest.h"
#include "ym/Literal.h"


BEGIN_NAMESPACE_YM

TEST(LiteralTest, empty_constr)
{
  Literal lit;

  EXPECT_EQ( kLiteralX, lit );
}

TEST(LiteralTest, simple_constr)
{
  VarId var(3);
  Literal lit(var);

  EXPECT_NE( kLiteralX, lit );

  VarId var1 = lit.varid();
  EXPECT_EQ( var, var1 );

  EXPECT_TRUE( lit.is_positive() );
  EXPECT_FALSE( lit.is_negative() );

  Literal lit_n = ~lit;
  EXPECT_NE( lit, lit_n );

  EXPECT_EQ( lit.varid(), lit_n.varid() );
  EXPECT_FALSE( lit_n.is_positive() );
  EXPECT_TRUE( lit_n.is_negative() );

  EXPECT_EQ( lit, lit.make_positive() );
  EXPECT_NE( lit, lit.make_negative() );

  EXPECT_NE( lit_n, lit_n.make_positive() );
  EXPECT_EQ( lit_n, lit_n.make_negative() );

}

END_NAMESPACE_YM

File: py-test/literal_test.py
#! /usr/bin/env python3

### @file literal_test.py
### @brief
### @author Yusuke Matsunaga (松永 裕介)
###
### Copyright (C) 2018 Yusuke Matsunaga
### All rights reserved.

import unittest
from pym_logic import VarId
from pym_logic import Literal


# Literal のテスト用クラス
class LiteralTest(unittest.TestCase) :

    def runTest(self) :
        self.test_empty_constructor()
        self.test_simple_constructor()

    def test_empty_constructor(self) :
        lit = Literal()
        self.assertFalse( lit.is_valid() )
        self.assertEqual( str(lit), '-X-' )

    def test_simple_constructor(self) :
        var = VarId(3)
        lit = Literal(var)

        var1 = lit.varid
        self.assertEqual( var, var1 )

        self.assertTrue( lit.is_positive() )
        self.assertFalse( lit.is_negative() )

        lit_n = ~lit
        self.assertNotEqual( lit, lit_n )

        self.assertEqual( lit.varid, lit_n.varid )
        self.assertFalse( lit_n.is_positive() )
        self.assertTrue( lit_n.is_negative() )

        self.assertEqual( lit, lit.make_positive() )
        self.assertEqual( lit_n, lit.make_negative() )

        self.assertEqual( lit, lit_n.make_positive() )
        self.assertEqual( lit_n, lit.make_negative() )

        self.assertEqual( str(lit), 'V_3' )
        self.assertEqual( str(lit_n), "V_3'" )



if __name__ == '__main__' :

    unittest.main()
